doc-surrealdb_versioned_docs/version-2.x/reference-guide/security-best-practices.mdx

```diff
@@ -0,0 +1,323 @@
+// Rust SDK
+// Do this:
+let name = "tobie"; // User-controlled input.
+let mut result = db
+    .query("CREATE person CONTENT name = $name")
+    .bind(("name", name))
+    .await?;
+```
+
+```rust
+// Rust SDK
+// highlight-next-line
+// Do NOT do this:
+let name = "tobie"; // User-controlled input.
+let mut result = db
+    .query(format!("CREATE person CONTENT name = {name}"))
+    .await?;
+```
+
+```jsx
+// JavaScript SDK
+// Do this:
+const name = "tobie"; // User-controlled input.
+const result = await db.query(
+	'CREATE person CONTENT name = "$name";',
+	{ name: name }
+);
+```
+
+```jsx
+// JavaScript SDK
+// highlight-next-line
+// Do NOT do this:
+const name = "tobie"; // User-controlled input.
+const result = await db.query('CREATE person CONTENT name = "' + name + '";');
+```
+
+## Content Safety
+
+Content generated by users and other untrusted parties will often be stored in SurrealDB and later rendered in an HTML page to be displayed. Regardless of SurrealDB, rendering untrusted content is the source of some dangerous pitfalls which can lead to [cross-site scripting](https://en.wikipedia.org/w/index.php?title=Cross-site_scripting&oldid=1232455342) attacks and other client-side code injection issues like [site defacement](https://en.wikipedia.org/w/index.php?title=Website_defacement&oldid=1231310592) or [clickjacking](https://en.wikipedia.org/w/index.php?title=Clickjacking&oldid=1227193298).
+
+When retrieving content that may be rendered in an HTML document, we strongly recommend that you use the [`string::html::encode`](https://surrealdb.com/docs/surrealdb/2.x/surrealql/functions/database/string#stringhtmlencode) function, which will encode any characters that have special meaning in HTML syntax (e.g. `<`, `>`, `&`...) into HTML entities (e.g. `&lt;`, `&gt;`, `&amp;`...) that will be rendered as the actual original character instead of interpreted as HTML syntax.
+
+**Example**
+
+```surql
+RETURN string::html::encode("<h1>Safe Title</h1><script>alert('XSS')</script><p>Safe paragraph. Not safe <span onload='logout()'>event</span>.</p>");
+
+['&lt;h1&gt;Safe&#32;Title&lt;&#47;h1&gt;&lt;script&gt;alert(&apos;XSS&apos;)&lt;&#47;script&gt;&lt;p&gt;Safe&#32;paragraph.&#32;Not&#32;safe&#32;&lt;span&#32;onload&#61;&apos;logout()&apos;&gt;event&lt;&#47;span&gt;.&lt;&#47;p&gt;']
+```
+
+If you absolutely require user-generated content to be rendered as HTML but still want to prevent users from injecting dangerous HTML into your page, you can use the [`string::html::sanitize`](https://surrealdb.com/docs/surrealdb/2.x/surrealql/functions/database/string#stringhtmlsanitize) function instead, which will keep all characters intact, so that the content can be interpreted as HTML syntax, while removing the specific syntax that is deemed dangerous. It is important to note that, although the set of accepted syntax is very conservative, sanitization is less safe that encoding and could potentially be bypassed due to a flaw in the function.
```

Comment:
```suggestion
If you absolutely require user-generated content to be rendered as HTML but still want to prevent users from injecting dangerous HTML into your page, you can use the [`string::html::sanitize`](/docs/surrealdb/2.x/surrealql/functions/database/string#stringhtmlsanitize) function instead, which will keep all characters intact, so that the content can be interpreted as HTML syntax, while removing the specific syntax that is deemed dangerous. It is important to note that, although the set of accepted syntax is very conservative, sanitization is less safe that encoding and could potentially be bypassed due to a flaw in the function.
```